lib/features/children/details/cubit/child_details_cubit.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:givt_app/features/children/details/models/profile_ext.dart';
+import 'package:givt_app/features/children/details/repositories/child_details_repository.dart';
+import 'package:givt_app/features/children/overview/models/profile.dart';
+
+part 'child_details_state.dart';
+
+class ChildDetailsCubit extends Cubit<ChildDetailsState> {
+  ChildDetailsCubit(this._childDetailsRepository)
+      : super(const ChildDetailsInitialState());
+
+  final ChildDetailsRepository _childDetailsRepository;
+
+  Future<void> fetchChildDetails(Profile profile) async {
+    emit(const ChildDetailsFetchingState());
+    try {
+      final response = await _childDetailsRepository.fetchChildDetails(profile);
+      emit(
+        ChildDetailsFetchedState(
+          profileDetails: response,
+        ),
+      );
+    } catch (error) {
+      emit(ChildDetailsErrorState(errorMessage: error.toString()));
```

Comment:
maybe it's helpful to log this to kibana. But why don't you catch on the GivtServerFailure? as you have a body and statusCode there

Reply:
yep, Loggit added and switched to GivtServerFailure instead of just error